### PR TITLE
[Feature] Type cast expression JIT

### PR DESCRIPTION
**Pull request by Moonm3n**
Related to https://github.com/StarRocks/starrocks/issues/28331, Implement arithmetic expression JIT.

## What type of PR is this:

This pr supports multiple types of cast operations. 
Include: `BOOLEAN`, `TYPE_TINYINT`, `TYPE_SMALLINT`, `TYPE_INT`, `TYPE_BIGINT`, `TYPE_LARGEINT`, `TYPE_FLOAT`, and `TYPE_DOUBLE`.

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exprs/cast_expr.cpp [hunk 1]
```diff
@@ -1127,6 +1134,47 @@ class VectorizedCastExpr final : public Expr {
...
+        auto* l = datums[0].value;
+
+        if constexpr (FromType == TYPE_JSON || ToType == TYPE_JSON) {
+            return Status::NotSupported("JIT casting does not support JSON");
+        } else if constexpr (lt_is_decimal<FromType> && lt_is_decimal<ToType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else if constexpr (lt_is_decimal<FromType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else if constexpr (lt_is_decimal<ToType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
```
Comment (fzhedu) [inline be/src/exprs/cast_expr.cpp:1153]: ||

File: be/src/exprs/cast_expr.cpp [hunk 2]
```diff
@@ -1127,6 +1134,47 @@ class VectorizedCastExpr final : public Expr {
...
+    bool is_compilable() const override {
+        return !AllowThrowException && IRHelper::support_jit(FromType) && IRHelper::support_jit(ToType);
+    }
+
+    StatusOr<LLVMDatum> generate_ir_impl(ExprContext* context, const llvm::Module& module, llvm::IRBuilder<>& b,
+                                         const std::vector<LLVMDatum>& datums) const override {
+        auto* l = datums[0].value;
+
+        if constexpr (FromType == TYPE_JSON || ToType == TYPE_JSON) {
+            return Status::NotSupported("JIT casting does not support JSON");
```
Comment (fzhedu) [inline be/src/exprs/cast_expr.cpp:1147]: is_complex_type, like json, object, map, array, struct, binary ? pls check all

File: be/src/exprs/cast_expr.cpp [hunk 3]
```diff
@@ -1127,6 +1134,47 @@ class VectorizedCastExpr final : public Expr {
...
+        } else if constexpr (lt_is_decimal<FromType> && lt_is_decimal<ToType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else if constexpr (lt_is_decimal<FromType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else if constexpr (lt_is_decimal<ToType>) {
+            return Status::NotSupported("JIT casting does not support decimal");
+        } else {
+            LLVMDatum datum(b);
```
Comment (fzhedu) [inline be/src/exprs/cast_expr.cpp:1155]: only numeric types except decimal?

File: be/src/exprs/cast_expr.cpp (hunk 2 above)
Comment (Moonm3n) [reply be/src/exprs/cast_expr.cpp:1147]: The ﻿is_compilable() function is designed to filter these types. Here, the ﻿if constexpr is utilized for the purpose of compilation.

File: be/src/exprs/cast_expr.cpp (hunk 1 above)
Comment (Moonm3n) [reply be/src/exprs/cast_expr.cpp:1153]: Done.

File: be/src/exprs/cast_expr.cpp (hunk 3 above)
Comment (Moonm3n) [reply be/src/exprs/cast_expr.cpp:1155]: Yep.

**Pull request closed without merging**